src/java_bytecode/java_class_loader.cpp

```diff
@@ -188,25 +187,26 @@ void java_class_loadert::read_jar_file(
   if(jar_map.find(file)!=jar_map.end())
     return;
 
-  jar_filet &jar_file=jar_pool(class_loader_limit, id2string(file));
-
-  if(!jar_file)
+  std::vector<std::string> filenames;
+  try
+  {
+    filenames=this->jar_pool(class_loader_limit, id2string(file)).filenames();
+  }
+  catch(const std::runtime_error&)
   {
     error() << "failed to open JAR file `" << file << "'" << eom;
     return;
   }
-
   debug() << "adding JAR file `" << file << "'" << eom;
 
   auto &jm=jar_map[file];
 
-  for(auto &jar_entry : jar_file.filtered_jar)
+  for(auto &file_name : filenames)
```

Comment:
I really tried hard to restrict changes to java_class loader. It's const implicitly.